src/model.ts

```diff
@@ -41,17 +41,24 @@ export class HubApiModel {
     // scraping entire database
     this.validateRequestScope(searchRequest);
 
-    const pagingStreams: PagingStream[] = await getBatchedStreams(searchRequest);
+    const limit = Number(request.query?.limit) || undefined;
+    
+    const pagingStreams: PagingStream[] = await getBatchedStreams(searchRequest, limit);
+    const pass: PassThrough =  new PassThrough({ objectMode: true });
+    return searchRequest.options.sortOrder && searchRequest.options.sortField 
```

Comment:
You just want to check for the existence of a sortField, since an absent sortOrder is actually still valid

Reply:
I agree. Made changes to only check for sortOrder.